FinniversKit/Sources/Components/TextField/TextField.swift

```diff
@@ -394,8 +407,20 @@ public class TextField: UIView {
 
     private func transition(to state: State) {
         layoutIfNeeded()
-        underlineHeightConstraint?.constant = state.underlineHeight
-
+        
+        if let dynamicBorder = self.textFieldDynamicBorder, dynamicBorder == true {
+            if state != .normal {
```

Comment:
What if we use a `switch` statement here instead?

```
switch state {
    case .normal: ...
    default: ...
}
self.layoutSubviews()
...
```

Reply:
FYI: Actually, we shouldn’t call `layoutSubviews` here. According to Apple we should never call it ourselves, but rather call `setNeedsLayout` or/and `layoutIfNeeded`. Details [here](https://developer.apple.com/documentation/uikit/uiview/1622482-layoutsubviews)